docs/concepts/models.md

```diff
@@ -181,36 +208,22 @@ print(m.model_dump())
 """
 ```
 
-For self-referencing models, see [postponed annotations](postponed_annotations.md#self-referencing-or-recursive-models).
-
-!!! note
-    When defining your models, watch out for naming collisions between your field name and its type, a previously defined model, or an imported library.
-
-    For example, the following would yield a validation error:
-    ```py test="skip"
-    from typing import Optional
-
-    from pydantic import BaseModel
-
+Self-referencing models are supported. For more details, see [postponed annotations](postponed_annotations.md#self-referencing-or-recursive-models).
 
-    class Boo(BaseModel):
-        int: Optional[int] = None
+## Rebuilding the model
 
-
-    m = Boo(int=123)  # errors
-    ```
-    An error occurs since the field  `int` is set to a default value of `None` and has the exact same name as its type, so both are interpreted to be `None`.
-
-## Rebuild model schema
-
-The model schema can be rebuilt using [`model_rebuild()`][pydantic.main.BaseModel.model_rebuild]. This is useful for building recursive generic models.
+When you define a model class in your code, Pydantic will analyze the body of the class to collect a number of information
+required to perform validation and serialization. Notably, the type annotations are evaluated to understand the valid types
+for each field (more information can be found in the [Architecture](../architecture.md) documentation). However, it might be
+that annotations refers to symbols not defined when the model class is being created. To circumvent this issue, the
+[`model_rebuild()`][pydantic.main.BaseModel.model_rebuild] method can be used:
```

Comment:
```suggestion
When you define a model class in your code, Pydantic will analyze the body of the class to collect information
required to perform validation and serialization. Notably, the model's type annotations are evaluated to understand the type constraints
for each field (more information can be found in the [Architecture](../architecture.md) documentation). However, it might be the case that some annotations refer to symbols not yet defined when the model class is being created. To circumvent this issue, the [`model_rebuild()`][pydantic.main.BaseModel.model_rebuild] method can be used:
```

Specifically, I don't think the "a number of" part is proper English. You could say "a variety of..." instead